File: RedSheet.DbEntities/Models/vEventAgendaTimingRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vEventAgendaTimingRecords", Schema = "dbo")]
	[Serializable]
	public partial class vEventAgendaTimingRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int EventAgendaTimingId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int NegotiationPhaseId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string PayOff { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Process { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectEventTimelineId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Purpose { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int SortOrder { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TacticsId { get; set; } 
 
	[Required]
	public TimeSpan Time { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Topic { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Trigger { get; set; } 

	public vEventAgendaTimingRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-event-agenda-timing-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vEventAgendaTimingRecord {
    constructor(vEventAgendaTimingRecord?: vEventAgendaTimingRecord )  {
        let properties = [ "eventAgendaTimingId", "negotiationPhaseId", "payOff", "process", "projectEventTimelineId", "purpose", "sortOrder", "tacticsId", "time", "topic", "trigger",];
        for (let property of properties)
            if (vEventAgendaTimingRecord && vEventAgendaTimingRecord[property])
                this[property] = vEventAgendaTimingRecord[property];
    }
 
	eventAgendaTimingId : number =   0 ;
 
    @range(1,2147483647)
	negotiationPhaseId : number =   undefined;
 
    @required()
    @maxLength(1000)
	payOff : string =   undefined;
 
    @required()
    @maxLength(1000)
	process : string =   undefined;
 
    @range(1,2147483647)
	projectEventTimelineId : number =   undefined;
 
    @required()
    @maxLength(1000)
	purpose : string =   undefined;
 
    @range(1,2147483647)
	sortOrder : number =   undefined;
 
    @range(1,2147483647)
	tacticsId : number =   undefined;
 
    @required()
	time : Date =   undefined;
 
    @required()
    @maxLength(1000)
	topic : string =   undefined;
 
    @required()
    @maxLength(1000)
	trigger : string =   undefined;


}
